lib/galaxy/datatypes/qiime2.py

```diff
@@ -76,6 +92,46 @@ def sniff(self, filename):
         return metadata and metadata['semantic_type'] == 'Visualization'
 
 
+class PredicateRemover(ast.NodeTransformer):
+    def visit_BinOp(self, node):
+        if isinstance(node.op, ast.Mod):
+            return PredicateRemover().visit(node.left)
+        return node
+
+
+class ReconstructExpression(ast.NodeVisitor):
+    expression = '%s'
+    tuple_count = 0
+    in_index = False
```

Comment:
These are static variables, they should be placed in an `__init__` to make them instance variables